dynamiq/memory/backend/base.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+
+from dynamiq.prompts import Message
+
+
+class Backend(ABC):
+    """Abstract base class for memory storage backends."""
+
+    @abstractmethod
+    def add(self, message: Message):
+        """Adds a message to the memory storage."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_all(self) -> list[Message]:
+        """Retrieves all messages from the memory storage."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def search(self, query: str, search_limit: int) -> list[Message]:
```

Comment:
I think we're missing a critical parameter here - filters. I should be able to filter by certain conditions before search. and I guess search query and filters can be allowed empty

Reply:
I will check the filters. 

The search query is empty — that makes sense.